### ISSUE TITLE
Add support for Tor HTTPTunnelPort

### ISSUE THREAD

**Issue by kzu** [joelverhagen/TorSharp#75]
See https://2019.www.torproject.org/docs/tor-manual.html.en#HTTPTunnelPort

**Comment by joelverhagen** [comment]
OMG how did I not know of this? If I understand correctly, does this completely eliminate the need for Privoxy?

**Comment by joelverhagen** [comment]
From https://gitweb.torproject.org/tor.git/plain/ChangeLog, it looks like this feature was added in this release:
> Changes in version 0.3.2.1-alpha - 2017-09-18

That's 2 years after I started the project. I guess I should have been staying on top of new features 😄. Thanks for the heads up!

**Comment by joelverhagen** [comment]
Okay, I've investigated the current state:

- .NET does not support SOCKS directly yet, but it looks like it will in .NET 6 via https://github.com/dotnet/runtime/pull/48883
- .NET does not use HTTP CONNECT if the destination is HTTP. HTTPS URLs work fine, per https://github.com/dotnet/runtime/issues/37690

Privoxy supports both HTTP proxy and HTTP CONNECT on the same port. This is why .NET works so well with it because Privoxy can handle both modes that .NET sends.

If you know all of the URLs you're hitting with `HttpClient` will be HTTPS, then you don't need Privoxy at all and you can do something like this: 

```csharp
            using (var proxy = new TorSharpProxy(settings))
            {
                var handler = new HttpClientHandler
                {
                    Proxy = new WebProxy(new Uri("http://localhost:" + settings.TorSettings.HttpTunnelPort))
                };

                using (handler)
                using (var httpClient = new HttpClient(handler))
                {
                    await proxy.ConfigureAndStartAsync();
                    Console.WriteLine(await httpClient.GetStringAsync("https://api.ipify.org"));
                }

                proxy.Stop();
            }
```

In short, for the purposes of this library, this is a great addition for extensibility purposes but Privoxy is still necessary for the non-HTTPS URL case.

**Comment by kzu** [comment]
Yep, was down that rabbit whole yesterday, hehe. Interesting distinction with HTTP vs HTTPS! That was tripping me up too!

**Comment by joelverhagen** [comment]
Release in TorSharp 2.8.0. Thanks again, @kzu!
https://www.nuget.org/packages/Knapcode.TorSharp/2.8.0
